### PR TITLE
PostHog based event tracking

### PR DESCRIPTION
**Pull request by ricklamers**
This PR replaces the old telemetry system and instead makes use of a self-hosted instance of PostHog to track anonymized user events in Orchest.

The tracking can still be disabled using the config.json user settings (which is also accessible in the settings pane).

Which events are tracked?
- Loading a view (the name of the view).
- Running a pipeline (through an experiment or through an interactive run).
- Alert/confirm dialog popups.
- Saving a pipeline (number of pipeline steps, and number of environments used).

### CODE REVIEW

File: services/orchest-webserver/app/app/analytics.py [hunk 1]
```diff
@@ -3,31 +3,104 @@
...
+
+    for _, step in steps.items():
+        step.pop("title", None)
+        step.pop("parameters", None)
+        step.pop("file_path", None)
+
+        # Capture environments for count
+        env = step.get("environment", "")
+        if env not in environments and len(env) > 0:
+            environments.append(env)
```
Comment (fruttasecca) [inline services/orchest-webserver/app/app/analytics.py:34]: I think environments could be a set.

File: services/orchest-webserver/app/app/analytics.py [hunk 2]
```diff
@@ -3,31 +3,104 @@
...
+            "project_size": project_size,
+            "run_type": run_type,
+        },
+    )
 
-        data = {"user_uuid": telemetry_uuid, "active": check_active(app)}
 
-        requests.post("https://analytics.orchest.io", json=data, timeout=1)
+def get_telemetry_uuid(app):
+    telemetry_uuid = ""
```
Comment (fruttasecca) [inline services/orchest-webserver/app/app/analytics.py:65]: minor thing, but I don't think the variable actually needs to be declared

File: services/orchest-webserver/app/requirements.txt [hunk 3]
```diff
@@ -14,4 +14,5 @@ nbformat
 docker>=4.3.0
 Werkzeug==0.16.0
 python-socketio[client]
+posthog
```
Comment (fruttasecca) [inline services/orchest-webserver/app/requirements.txt:17]: should we fix the version?

**Review by fruttasecca — COMMENTED** [review]
lgtm

Since Orchest is project centric, should we also track project level information? Number of pipelines, number of environments, number of run experiments,  maybe also number of projects.

**Comment by ricklamers** [comment]
> lgtm
> 
> Since Orchest is project centric, should we also track project level information? Number of pipelines, number of environments, number of run experiments, maybe also number of projects.

Let's keep this for future expansion if necessary. I don't want to track too many things right now.

**Comment by yannickperrenet** [comment]
Looks good! Although it is probably a good idea to add specific comments in the code explaining why we track certain characteristics.

**Review by yannickperrenet — APPROVED** [review]

**Pull request merged**